examples/cityscapes/leap_binder.py

```diff
@@ -0,0 +1,308 @@
+    CATEGORIES_id_no_background
+from project_config import IMAGE_STD, IMAGE_MEAN, IMAGE_SIZE, BACKGROUND_LABEL, SMALL_BBS_TH
+from utils_all.general_utils import extract_bounding_boxes_from_instance_segmentation_polygons, \
+    bb_array_to_object, get_predict_bbox_list
+from visualizers.visualizers import bb_decoder, gt_bb_decoder, bb_car_gt_decoder, bb_car_decoder
+
+from code_loader import leap_binder
+from code_loader.contract.datasetclasses import PreprocessResponse
+from code_loader.contract.enums import (
+    DatasetMetadataType,
+    LeapDataType
+)
+
+# ----------------------------------------------------data processing--------------------------------------------------
+def load_cityscapes_data_leap() -> List[PreprocessResponse]:
+    all_images, all_gt_images, all_gt_labels, all_gt_labels_for_bbx, all_file_names, all_metadata, all_cities =\
+        load_cityscapes_data()
+
+    # train_len = len(all_images[0])
+    # val_len = len(all_images[1])
+    # test_len = len(all_images[2])
+
+    train_len = 700
+    val_len = 100
+    test_len = 200
+
+    responses = [PreprocessResponse(length=train_len, data={
+                "image_path": all_images[0],
+                "subset_name": "train",
+                "gt_path": all_gt_labels[0],
+                "gt_bbx_path": all_gt_labels_for_bbx[0],
+                "gt_image_path": all_gt_images[0],
+                "real_size": train_len,
+                "file_names": all_file_names[0],
+                "cities": all_cities[0],
+                "metadata": all_metadata[0],
+                "dataset": ["cityscapes"]*train_len}),
+                PreprocessResponse(length=val_len, data={
+                "image_path": all_images[1],
+                "subset_name": "val",
+                "gt_path": all_gt_labels[1],
+                "gt_bbx_path": all_gt_labels_for_bbx[1],
+                "gt_image_path": all_gt_images[1],
+                "real_size": val_len,
+                "file_names": all_file_names[1],
+                "cities": all_cities[1],
+                "metadata": all_metadata[1],
+                "dataset": ["cityscapes"]*val_len}),
+                PreprocessResponse(length=test_len, data={
+                "image_path": all_images[2],
+                "subset_name": "test",
+                "gt_path": all_gt_labels[2],
+                "gt_bbx_path": all_gt_labels_for_bbx[2],
+                "gt_image_path": all_gt_images[2],
+                "real_size": test_len,
+                "file_names": all_file_names[2],
+                "cities": all_cities[2],
+                "metadata": all_metadata[2],
+                "dataset": ["cityscapes"] * test_len})]
+    return responses
+
+#------------------------------------------input and gt------------------------------------------
+
+def non_normalized_image(idx: int, data: PreprocessResponse) -> np.ndarray:
+    data = data.data
+    cloud_path = data['image_path'][idx%data["real_size"]]
```

Comment:
Why do we need this %? we usually do this when training to make the epoch shorter.
Consider removing if not important